src/ng/filter/filter.js

```diff
@@ -45,8 +45,9 @@
  *     The final result is an array of those elements that the predicate returned true for.
  *
  * @param {function(actual, expected)|true|false} [comparator] Comparator which is used in
- *     determining if the expected value (from the filter expression) and actual value (from
- *     the object in the array) should be considered a match.
+ *     determining if matches returned by the argument expression should be considered a match based
```

Comment:
The matches are not _returned_ by the `expression` argument. How about something like:

> determining if values retrieved using `expression` should be...